test/cases/enum.ts

```diff
@@ -6,7 +6,8 @@ export var schema = {
       "enum": ["a", "b", "c"]
     },
     "bar": {
-      "enum": [1, 2, 3]
+      "enum": [1, 2, 3],
+      "tsEnumNames": ["One","Two","Three"]
```

Comment:
I think I agree with you on principle starting from a clean slate, however this implementation is informed by my current use case, which is generating the same contract objects using both this library and jsonschema2pojo.  

Also I think the schema definition implies that the values of `enum` are the possible literal values that might occur in json passed over the wire... which in our case are technically literal integers if we're talking about TS enums.

In keeping with this, the jsonschema2pojo implementation uses `enum` as the literal values (strings, integers [I had to add support for that in a PR]) and uses `javaEnumNames` as the symbolic names of the members, since the symbolic names are not what's serialized per the schema.  So what I really need here is interoperability, and I think this implementation, though ugly, does conform to the json-schema intent.